prov/efa/test/efa_unit_test_info.c

```diff
@@ -98,3 +98,51 @@ void test_info_open_ep_with_api_1_1_info()
 	err = fi_close(&fabric->fid);
 	assert_int_equal(err, 0);
 }
+
+void test_info_get_shm_info()
+{
+	struct fi_info *hints, *info;
+	struct fid_fabric *fabric = NULL;
+	struct fid_domain *domain = NULL;
+	int err;
+	struct efa_domain *efa_domain;
+
+	hints = efa_unit_test_alloc_hints(FI_EP_DGRAM);
+
+	err = fi_getinfo(FI_VERSION(1, 14), NULL, NULL, 0ULL, hints, &info);
+	assert_int_equal(err, 0);
+
+	err = fi_fabric(info->fabric_attr, &fabric, NULL);
+	assert_int_equal(err, 0);
+
+	/* Hacking the info to make it have FI_HMEM cap */
+	info->caps |= FI_HMEM;
```

Comment:
Is it possible for shm to fail initialization if no HMEM interfaces are found?

Reply:
See the newest revision